### PR TITLE
Delete Artifacts based on the Properties

### PR DESCRIPTION
**Pull request by Imranonline**
Delete Artifacts based on the properties on the folders and artifacts.

### CODE REVIEW

**Comment by elioengcomp** [comment]
@Imranonline thank you for the contribution.

Besides the plugin file, we also need a `README.md` with at least the following content:

- Description
- Features
- Installation
- Usage

You can use other plugins README files as a reference.

Finally, please [sign the CLA](https://github.com/JFrogDev/artifactory-user-plugins/blob/master/CONTRIBUTING.md) so we can merge your PR.

**Comment by Imranonline** [comment]
Thank you for the suggestion @elioengcomp , I have added the Readme file now

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        depreciateData("repositoryname-local")
+    }
+}
+def depreciateData(repoName){
+    
+    log.info "Starting data depreciation job for repository " + repoName // This will log in the LogFile
+    
+    // search given repo name for folders with property DataDeprecation = true 
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: This search is looking for items in all repositories. Both `repoName` and `repoRootPath` variables are not being used. I don't believe that is the desired behavior. If you want to run the search in a specific repository add the repository name as an argument to the `itemsByProperties` method.

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+}
+      
+    }
+}
+/**
+    Deprecate files by Module name and major version, removing by build number
+    Keep last x number of semantic versions of an Artifact
+    args is assumed to be a single integer value (but as a string) for number of files to keep
+*/
+def moduleMajorVersion(RepoPath repoPath, repoName,String[] args){
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: `args` is always hosting a single value. There is no need for it to be a list. Also, the parameter name should indicate the function of the parameter.

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+args.each{
+    log.info "Depreciating data using method moduleMajorVersion and args " + it
+    println "Depreciating data using method moduleMajorVersion and args " + it
+    numToKeep=it
+}
+    println "Starting data depreciation job for repository " + repoName
+    log.info "Starting data depreciation job for repository " + repoName
+    // search given repo name for folders with property DataDeprecation = true
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: You already have the path to cleanup at `repoPath` parameter. Why do you need to perform this search again?

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+def int numToKeep =0
+args.each{
+    log.info "Depreciating data using method moduleMajorVersion and args " + it
+    println "Depreciating data using method moduleMajorVersion and args " + it
+    numToKeep=it
+}
+    println "Starting data depreciation job for repository " + repoName
+    log.info "Starting data depreciation job for repository " + repoName
+    // search given repo name for folders with property DataDeprecation = true
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: Both `repoName` and `repoRootPath` variables are not being used.

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    for(RepoPath rPath : deprecatePaths){
+        println "Depreciating data for path " + rPath
+        log.info "Depreciating data for path " + rPath
+         def parent = rPath.parent
+       println "Parent Path " + parent
+
+       folderParent= repositories.getChildren(rPath)
+      for (child in folderParent) {
+         RepoPath childRepoPath = child.getRepoPath()
+        def artifactVersion = repositories.getProperty(childRepoPath,"build.number")
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: You are getting the artifact version from the `build.number` property and assuming that it has the `x.x.x.x` format. What happens if that is not true?

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy (hunk 2 above)
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: This method's name should describe what it does.

File: cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    println "Depreciating data using method moduleMajorVersion and args " + it
+    numToKeep=it
+}
+    println "Starting data depreciation job for repository " + repoName
+    log.info "Starting data depreciation job for repository " + repoName
+    // search given repo name for folders with property DataDeprecation = true
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
+    List<ItemInfo> folderParent 
+    List<String> fillVersions =[""]
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy]: Using a Set would make it easier to handle duplicates.

File: cleanup/DeleteBasedonProperties/Readme.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+Artifacts delete based on artifact properties User Plugin
+=====================================
+##Description:
+This plugin deletes builds based on properties defined on repositories . It can be run automatically as a scheduled job.
+
+##Features:
+This plugin Takes the Parameters from the Artifact/Repository
+
+It will scan all the repositories marked with property as `DataDeprecation'= 'true' `
+It will then look for property called `DataDeprecationMethod = 'version : x' `
```
Comment (elioengcomp) [inline cleanup/DeleteBasedonProperties/Readme.md]: You probably don't need two properties. Just by checking property `DataDeprecationMethod` is enough to identify that that path must be cleaned. You should also change the property name to be in sync with the plugin and change the value to a single integer with the desired number of versions to keep. Suggestion: `cleanOldVersions.versionsToKeep = x`

**Comment by yashprit-jfrog** [comment]
We are minimizing our efforts on plugin in favor for [Jfrog Worker](https://jfrog.com/help/r/jfrog-platform-administration-documentation/workers) & not taking request for new plugin.  

Closing this PR due to inactivity.

**Pull request closed without merging**